Composer/packages/client/src/pages/knowledge-base/table-view.tsx

```diff
@@ -269,15 +275,25 @@ const TableView: React.FC<TableViewProps> = (props) => {
 
   const onSubmitEditKB = async ({ name }: { name: string }) => {
     if (!editQnAFile) return;
-    const newId = `${name}.source`;
-    await actions.renameQnAKB({ id: editQnAFile.id, name: newId, projectId: actualProjectId });
+    const newSourceId = `${name}.source`;
+    for (let i = 0; i < languages.length; i++) {
+      await actions.renameQnAKB({
+        id: `${getBaseName(editQnAFile.id)}.${languages[i]}`,
+        name: name,
+        projectId: actualProjectId,
+      });
+    }
```

Comment:
This looks like another `await Promise.all( languages.map( (language) => ... ) )` kind of block.